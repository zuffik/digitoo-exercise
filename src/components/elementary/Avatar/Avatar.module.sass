.avatar
  display: inline-block
  border-radius: 50%
  overflow: hidden
