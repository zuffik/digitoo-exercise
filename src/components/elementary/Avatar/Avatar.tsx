import * as React from "react";
import classNames from "classnames";
import styles from "./Avatar.module.sass";

interface Props
  extends Omit<React.ImgHTMLAttributes<HTMLImageElement>, "width" | "height"> {
  size?: number;
}

export const Avatar: React.FC<Props> = (props: Props) => {
  const size = props.size || 48;
  return (
    <img
      {...props}
      alt={props.alt}
      className={classNames(props.className, styles.avatar)}
      width={size}
      height={size}
    />
  );
};
