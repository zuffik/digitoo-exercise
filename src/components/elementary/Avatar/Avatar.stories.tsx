import * as React from 'react';
import {Avatar} from './Avatar';

export default {
  title: 'Elementary',
};

export const avatar = () => <Avatar src="https://picsum.photos/48" alt="avatar" />;
