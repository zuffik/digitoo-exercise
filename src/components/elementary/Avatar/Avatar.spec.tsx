import * as React from 'react';
import {render} from '@testing-library/react'
import {Avatar} from "./Avatar";

describe('<Avatar/>', () => {
    it('should render', () => {
        const {asFragment, getByAltText} = render(<Avatar alt="id" src="image.png"/>);
        expect(getByAltText('id')).toBeInTheDOM();
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render with specific size', () => {
        const {asFragment, getByAltText} = render(<Avatar size={50} alt="id" src="image.png"/>);
        expect(getByAltText('id')).toHaveAttribute('width', '50');
        expect(getByAltText('id')).toHaveAttribute('height', '50');
        expect(asFragment()).toMatchSnapshot();
    });
});
