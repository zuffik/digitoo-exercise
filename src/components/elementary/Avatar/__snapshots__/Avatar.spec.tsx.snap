// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> should render 1`] = `
<DocumentFragment>
  <img
    alt="id"
    class="avatar"
    height="48"
    src="image.png"
    style="min-width: 48px;"
    width="48"
  />
</DocumentFragment>
`;

exports[`<Avatar/> should render with specific size 1`] = `
<DocumentFragment>
  <img
    alt="id"
    class="avatar"
    height="50"
    size="50"
    src="image.png"
    style="min-width: 50px;"
    width="50"
  />
</DocumentFragment>
`;
